src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -54,19 +55,22 @@
   private static final String TIMESTAMP = "timestamp";
   private static final String TITLE = "title";
   private static final String SENTIMENT = "sentiment";
-  private static final String UPVOTES = "upvotes";
+  private static final String LIKES = "likes";
   private static final String URL = "url";
-  private static List<String> threadTitles = new ArrayList<String>();
-  private static List<Double> threadSentiments = new ArrayList<Double>();
-  private static List<Integer> threadUpvotes = new ArrayList<Integer>();
-  private static List<String> threadUrls = new ArrayList<String>();
-  private final JSONObject threadInfoList = new JSONObject();
+  private static final String NUMOFPAGES = "numOfPages";
+
+  private final JSONObject threadInfo = new JSONObject();
   private final Gson gson = new Gson();
   private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
   private final Analyze analyze = new Analyze();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    List<String> threadTitles = new ArrayList<String>();
+    List<Double> threadSentiments = new ArrayList<Double>();
+    List<Integer> threadLikes = new ArrayList<Integer>();
+    List<String> threadUrls = new ArrayList<String>();
```

Comment:
I still think these should be replaced with a single entity instead.

Could you define your UI model and populate it instead?

For example, using [AutoValue](https://github.com/google/auto/blob/master/value/userguide/index.md) (which is a very nice way to create immutable data classes, but entirely optional):
```java
@AutoValue
class AnalyzedPost {
  String title();
  double sentiment();
  int likes();
  String url();

  public static AnalyzedPost create(String title, double sentiment, int likes, String url) {
    return new AutoValue_AnalyzedPost(title, sentiment, likes, url);
  }
}
```

Reply:
If I did this, wouldn't I need to create a list of the class and if I did create a list wouldn't I have to find a new way to set up the json?

Reply:
This can be done in a followup PR